Applications/Spire/Include/Spire/Ui/DecimalInputWidget.hpp

```diff
@@ -0,0 +1,65 @@
+#ifndef SPIRE_DECIMAL_INPUT_WIDGET_HPP
+#define SPIRE_DECIMAL_INPUT_WIDGET_HPP
+#include "Spire/Ui/NumericInputWidget.hpp"
+
+namespace Spire {
+
+  //! Represents a widget for displaying and modifying decimal numbers.
+  class DecimalInputWidget : public QWidget {
```

Comment:
Rename this widget to ```DecimalSpinBox``` and inherit from ```QDoubleSpinBox``` or ```QAbstractSpinBox```.

Reply:
Is there an advantage to repeating the implementation for each type? Any change in the UX flow will have to be repeated for each implementation.

Reply:
Use the ```RealSpinBox``` to factor out the vast majority of the implementation and delegate calls back and forth.

Reply:
That makes more sense but I still don't understand the advantage of treating these inputs as spin boxes instead of regular widgets. A plus is that the QSpinBox and QDoubleSpinBox provide a valueChanged signal, but this brings up issues with having two spinboxes and having to overwrite the inherited spin box with the member spin box.